apps/ui/pages/reservation-unit/[id].tsx

```diff
@@ -377,54 +413,143 @@ const ReservationUnit = ({
   isPostLogin,
   mapboxToken,
   apiBaseUrl,
+  searchDuration,
+  searchDate,
+  searchTime,
 }: PropsNarrowed): JSX.Element | null => {
   const { t, i18n } = useTranslation();
   const router = useRouter();
   const now = useMemo(() => new Date(), []);
   const isMobile = useMedia(`(max-width: ${breakpoints.m})`, false);
 
-  const [focusDate, setFocusDate] = useState(new Date());
   const [calendarViewType, setCalendarViewType] = useState<WeekOptions>("week");
-  const [initialReservation, setInitialReservation] =
-    useState<PendingReservation | null>(null);
-  const [isReserving, setIsReserving] = useState(false);
   const [isDialogOpen, setIsDialogOpen] = useState(false);
   const [errorMsg, setErrorMsg] = useState<string | null>(null);
-  const [shouldUnselect, setShouldUnselect] = useState(0);
-  const [storedReservation, , removeStoredReservation] =
-    useLocalStorage<PendingReservation>("reservation");
 
   const calendarRef = useRef<HTMLDivElement>(null);
-  const openPricingTermsRef = useRef<HTMLAnchorElement>(null);
+
   const hash = router.asPath.split("#")[1];
 
-  const isClientATouchDevice = isTouchDevice();
+  const reservableTimeSpans = useMemo(
+    () => filterNonNullable(reservationUnit?.reservableTimeSpans),
+    [reservationUnit?.reservableTimeSpans]
+  );
+  const todaysTimeSpans = reservableTimeSpans.filter(
+    (span) => span.startDatetime && isToday(new Date(span.startDatetime))
+  );
+  const searchUIDate = fromUIDate(searchDate ?? "");
+  // TODO: plug in query parameters
```

Comment:
still?

Reply:
Well not exactly, but the form state should eventually only exist in the query parameters (i.e. the controlled inputs should also update the query params in the URL). However, this is not the optimal place to have a reminder about that entire refactoring need. Removed the comment.